### DISCUSSION TITLE
Internationalized Routing defaultLocale not working

### DISCUSSION THREAD

**Discussion by IvonaPilcevic** [vercel/next.js#20449] [Help]
I upgraded nextjs to 10 so I could use Internationalized Routing. This is how my nex.config.js file is looking: `module.exports = {
  i18n: {
    locales: ['rs', 'en'],
    defaultLocale: 'rs',
}
}
`

and everything is working fine, except one thing and that is when i open app it's always using en, instead of rs. When I change the language to rs it works fine, myApp.com/ and for en is myApp.com/en. But it just looks like defaultLocale isn't working like it should and I don't know where the problem is.

**Comment by ijjk** [comment]
Hi, can you provide a minimal reproduction for what is not working? Are you expecting the `rs` locale to always be used? If so this isn't the case, `defaultLocale` is what is used when visiting a non-locale prefixed URL e.g. `/hello` as long as your browser doesn't prefer a different locale. If your browser prefers the `en` locale you will be redirected to `/en` when visiting `/`.

**Reply by IvonaPilcevic** [reply]
Ok, so is there a way to ignore what my browser prefers and just automatically open app on rs language? And then in language switcher if you want you can change language to en. But I just need my app to be on rs by default.
Thank you!

**Reply by ijjk** [reply]
You can disable `localeDetection` in your `next.config.js` ([see documentation here](https://nextjs.org/docs/advanced-features/i18n-routing#disabling-automatic-locale-detection)) which will prevent redirecting based on the user's browser preference and then add a language switcher that updates which locale is being shown instead using the `defaultLocale` when visiting `/`.

**Reply by IvonaPilcevic** [reply]
Thank you so much, I'll try this!
